### PR TITLE
DOC: clean io.rst codeblocks for PEP8

### LINKED ISSUE #20508: DOC: io.rst uses deprecated Index.summary
https://github.com/pandas-dev/pandas/blob/cdfce2b0ad99f7faad57cc5247cf33aab5725bed/doc/source/io.rst (line 3979 at the moment).

Should probably just change it to `c`.

### PR DESCRIPTION
**Pull request by pulkitmaloo**
Checklist for the pandas documentation sprint (ignore this if you are doing
an unrelated PR):

- [x] PR title is "DOC: update the <your-function-or-method> docstring"
- [ ] The validation script passes: `scripts/validate_docstrings.py <your-function-or-method>`
- [x] The PEP8 style check passes: `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] The html version looks good: `python doc/make.py --single <your-function-or-method>`
- [ ] It has been proofread on language by another sprint participant

Please include the output of the validation script below between the "```" ticks:

```

```

If the validation script still gives errors, but you think there is a good reason
to deviate in this case (and there are certainly such cases), please state this
explicitly.

Checklist for other PRs (remove this part if you are doing a PR for the pandas documentation sprint):

- [x] closes #20508
- [ ] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: doc/source/io.rst [hunk 1]
```diff
@@ -921,7 +921,7 @@ Note that performance-wise, you should try these methods of parsing dates in ord
 1. Try to infer the format using ``infer_datetime_format=True`` (see section below).
 
 2. If you know the format, use ``pd.to_datetime()``:
-   ``date_parser=lambda x: pd.to_datetime(x, format=...)``.
+   ``date_parser = lambda x: pd.to_datetime(x, format=...)``.
```
Comment (jorisvandenbossche) [inline doc/source/io.rst]: This one is not necessarily correct, as `date_parser` is a keyword, and for keywords in function calls no spaces ...

File: doc/source/io.rst [hunk 2]
```diff
@@ -3975,11 +3976,12 @@ Sometimes you want to get the coordinates (a.k.a the index locations) of your qu
...
-   df_coord = pd.DataFrame(np.random.randn(1000,2),index=pd.date_range('20000101',periods=1000))
-   store.append('df_coord',df_coord)
-   c = store.select_as_coordinates('df_coord','index>20020101')
-   c.summary()
-   store.select('df_coord',where=c)
+   df_coord = pd.DataFrame(np.random.randn(1000, 2), 
+                           index=pd.date_range('20000101', periods=1000))
+   store.append('df_coord', df_coord)
+   c = store.select_as_coordinates('df_coord', 'index > 20020101')
+   c
```
Comment (jorisvandenbossche) [inline doc/source/io.rst]: this change of c.summary() to c was on purpose? (not familiar with HDFStore)

**Review by jorisvandenbossche — COMMENTED** [review]
Thanks, this is very welcome. Added a minor comment

File: doc/source/io.rst (hunk 2 above)
Comment (jreback) [reply doc/source/io.rst]: .summary() was recently deprecate so this is ok

Comment (jorisvandenbossche) [reply doc/source/io.rst]: Ah, the return value is an index, not HFDStore, OK +1

**Comment by jreback** [comment]
thanks @pulkitmaloo (fixed a few typos)

**Comment by pulkitmaloo** [comment]
Thanks for fixing the typos and accepting the PR. 
I found some more doc's code-blocks are not PEP8 compliant. I will work on it and create a new PR.

**Pull request merged**